website/search/elastic_search.py

```diff
@@ -653,13 +651,25 @@ def update_file(file_, index=None, delete=False):
     target = file_.target
 
     # TODO: Can remove 'not file_.name' if we remove all base file nodes with name=None
-    file_node_is_qa = bool(
-        set(settings.DO_NOT_INDEX_LIST['tags']).intersection(file_.tags.all().values_list('name', flat=True))
-    ) or bool(
-        set(settings.DO_NOT_INDEX_LIST['tags']).intersection(target.tags.all().values_list('name', flat=True))
-    ) or any(substring in target.title for substring in settings.DO_NOT_INDEX_LIST['titles'])
-    if not file_.name or not target.is_public or delete or file_node_is_qa or getattr(target, 'is_deleted', False) or getattr(target, 'archiving', False) or target.is_spam or (
-            target.spam_status == SpamStatus.FLAGGED and settings.SPAM_FLAGGED_REMOVE_FROM_SEARCH):
+
+    tags_not_to_index = set(settings.DO_NOT_INDEX_LIST['tags'])
+    file_tags = file_.tags.all().values_list('name', flat=True)
+    target_tags = file_.tags.all().values_list('name', flat=True)
+
+    file_tag_in_do_not_index = bool(tags_not_to_index.intersection(file_tags))
+    target_tag_in_do_not_index = bool(tags_not_to_index.intersection(target_tags))
+
+    if hasattr(target, 'title'):
+        part_of_tag_in_target_title = any(substring in target.title for substring in settings.DO_NOT_INDEX_LIST['titles'])
+    else:
+        part_of_tag_in_target_title = False
+
+    is_public = getattr(target, 'is_public', False) or file_.is_quickfile
```

Comment:
Third time I've seen getattr(target, 'is_public', False) - let's move to FileTargetMixin?  Will be true on quickfiles.